nexus/src/app/vpc.rs

```diff
@@ -6,12 +6,11 @@
 
 use crate::app::sagas;
 use crate::external_api::params;
+use nexus_capabilities::FirewallRules;
```

Comment:
> Suppose I'm adding "app-level logic" to Nexus. Should I add it to nexus/capabilities? or to nexus/src/app? What do you think should be the dividing line between the two?

Good question; I'm not sure! I could see arguments for at least:

* Always add to `nexus/capabilities`; this lets us keep it organized and makes it accessible to sagas (assuming we rework them to use this instead of taking an `Arc<Nexus>`) and RPWs
* Just keep adding to `nexus/src/app` as we have been. If someone needs functionality in an RPW, they can move it to `nexus/capabilities`.

I'm slightly partial to the first because it feels a little cleaner and encourages "only use the parts of `Nexus` you really need" (since via `Base` you only have access to the datastore, you'd have to add methods to get at other things). But the second seems perfectly reasonable and maybe an easier transition.

> I grok that the capabilities crate allows "things without an Arc<Nexus> to access the APIs, so is this crate basically treated like "another flavor of pub"?

Yes

Reply:
> If we did this, in the limit, would it still solve the problem? Or would we wind up with stuff inside this package that depends on other stuff inside this package that it shouldn't know about? I guess we could always create traits for only the limited functionality they need and this should be fine.

The goal is that any functionality that moves here only has access to data it _needs_, and that those dependencies are obvious (such as `FirewallRules` requiring `SledAgent`, since it has to construct sled-agent clients). We could certainly end up with weird cross-dependencies, but I think trying to split it up like this would make that obvious, and possibly painful enough that we'd take steps to avoid it.

> To that point, could the contents of nexus-capabilities live in the main omicron-nexus package?

I don't think so; that would force `blueprint-execution` (and any other smaller crates that want to use Nexus-lite functionality) to depend on `omicron-nexus`, creating a cycle.